### ISSUE TITLE
Web startup loop from error: telebot.apihelper.ApiException

### ISSUE THREAD

**Issue by t413** [TheSpaghettiDetective/TheSpaghettiDetective#151]
Fresh install, worked _great_. Hats off to your incredible efforts! Got SSL set up in order to get telegram working (before seeing this [gold mine of a service (localtunnel)](https://localtunnel.github.io/www/)). I needed some more compute power so changed some settings and rebooted.. booted back up the web task stuck in a loop. 

Googling around I found [this possiblity on stackoverflow](https://stackoverflow.com/questions/41172749/error-404-while-accessing-api-method-on-telegram-bot).. I'll also investigate.

Here's the results of `docker-compose log web`

```
web_1     | /usr/local/lib/python3.6/site-packages/daphne/server.py:13: UserWarning: Something has already installed a non-asyncio Twisted reactor. Attempting to uninstall it; you can fix this warning by importing daphne.server early in your codebase or finding the package that imports Twisted and importing it later on.
web_1     |   UserWarning,
web_1     |
web_1     | 0 static files copied to '/app/static_build', 220 unmodified, 485 post-processed.
web_1     | /usr/local/lib/python3.6/site-packages/daphne/server.py:13: UserWarning: Something has already installed a non-asyncio Twisted reactor. Attempting to uninstall it; you can fix this warning by importing daphne.server early in your codebase or finding the package that imports Twisted and importing it later on.
web_1     |   UserWarning,
web_1     | Traceback (most recent call last):
web_1     |   File "manage.py", line 15, in <module>
web_1     |     execute_from_command_line(sys.argv)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
web_1     |     utility.execute()
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
web_1     |     self.fetch_command(subcommand).run_from_argv(self.argv)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 316, in run_from_argv
web_1     |     self.execute(*args, **cmd_options)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 350, in execute
web_1     |     self.check()
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 379, in check
web_1     |     include_deployment_checks=include_deployment_checks,
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/commands/migrate.py", line 60, in _run_checks
web_1     |     issues.extend(super()._run_checks(**kwargs))
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 366, in _run_checks
web_1     |     return checks.run_checks(**kwargs)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/checks/registry.py", line 71, in run_checks
web_1     |     new_errors = check(app_configs=app_configs)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/checks/urls.py", line 13, in check_url_config
web_1     |     return check_resolver(resolver)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/core/checks/urls.py", line 23, in check_resolver
web_1     |     return check_method()
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/urls/resolvers.py", line 396, in check
web_1     |     for pattern in self.url_patterns:
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/utils/functional.py", line 37, in __get__
web_1     |     res = instance.__dict__[self.name] = self.func(instance)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/urls/resolvers.py", line 533, in url_patterns
web_1     |     patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/utils/functional.py", line 37, in __get__
web_1     |     res = instance.__dict__[self.name] = self.func(instance)
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/urls/resolvers.py", line 526, in urlconf_module
web_1     |     return import_module(self.urlconf_name)
web_1     |   File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
web_1     |     return _bootstrap._gcd_import(name[level:], package, level)
web_1     |   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
web_1     |   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
web_1     |   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
web_1     |   File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
web_1     |   File "<frozen importlib._bootstrap_external>", line 678, in exec_module
web_1     |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
web_1     |   File "/app/config/urls.py", line 21, in <module>
web_1     |     path('', include('app.urls')),
web_1     |   File "/usr/local/lib/python3.6/site-packages/django/urls/conf.py", line 34, in include
web_1     |     urlconf_module = import_module(urlconf_module)
web_1     |   File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
web_1     |     return _bootstrap._gcd_import(name[level:], package, level)
web_1     |   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
web_1     |   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
web_1     |   File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
web_1     |   File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
web_1     |   File "<frozen importlib._bootstrap_external>", line 678, in exec_module
web_1     |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
web_1     |   File "/app/app/urls.py", line 3, in <module>
web_1     |     from . import views
web_1     |   File "/app/app/views.py", line 20, in <module>
web_1     |     from .telegram_bot import bot_name
web_1     |   File "/app/app/telegram_bot.py", line 21, in <module>
web_1     |     bot_name = bot.get_me().username
web_1     |   File "/usr/local/lib/python3.6/site-packages/telebot/__init__.py", line 478, in get_me
web_1     |     result = apihelper.get_me(self.token)
web_1     |   File "/usr/local/lib/python3.6/site-packages/telebot/apihelper.py", line 93, in get_me
web_1     |     return _make_request(token, method_url)
web_1     |   File "/usr/local/lib/python3.6/site-packages/telebot/apihelper.py", line 56, in _make_request
web_1     |     return _check_result(method_name, result)['result']
web_1     |   File "/usr/local/lib/python3.6/site-packages/telebot/apihelper.py", line 75, in _check_result
web_1     |     raise ApiException(msg, method_name, result)
web_1     | telebot.apihelper.ApiException: A request to the Telegram API was unsuccessful. The server returned HTTP 404 Not Found. Response body:
web_1     | [b'{"ok":false,"error_code":404,"description":"Not Found"}']
```

**Comment by t413** [comment]
Found that telegram functionally hinges on having the token (telegram_bot.py:13) so I tried commenting-out my key and the web server started a-ok. 

Uncommented the key and ran `docker-compose restart` and it restarted perfectly.. and it _seems_ telegram is enabled. I do wish there were a 'test-send telegram notification' button in the user preferences.

Still don't know why I get the error on system-reboot. If you'd like me to test system-rebooting again to debug the issue more I'm happy to.

**Comment by t413** [comment]
Found my issue and resolved it. It had to do with the full API key not being used. The token _should_ look like `123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11` with the colon and all. I'd had trouble getting things working earlier and tried without the `123456:` part __and it worked__. Registered, received notifications and everything. Except that now it stopped working after a full stop-start. Go figure.

Perhaps some verification on that API key? And perhaps a try block around the get_me() call so it doesn't keep bringing down the whole server?

To figure this out I added this to enable verbosity in web/app/telegram_bot.py:

```
import telebot
telebot.logger.setLevel(logging.DEBUG)
```
Then compared the api call to [the telegram docs](https://core.telegram.org/bots/api#available-methods) to realize the API key was different. Added in the whole key and re-registered my user and it worked like a charm.
